src/components/FeaturePanel/PublicTransport/requestRoutes.ts

```diff
@@ -0,0 +1,43 @@
+import { fetchText } from '../../../services/fetch';
+
+interface LineInformation {
+  ref: string;
+  colour: string | undefined;
+}
+
+export async function requestLines(
+  featureType: 'node' | 'way' | 'relation',
+  id: number,
+  radius = 150,
+) {
+  // use the overpass api to request the lines in
+  const overpassQuery = `[out:csv(ref, colour; false; ';')];
+    ${featureType}(${id})->.center;
+    node(around.center:${radius})["public_transport"="stop_position"] -> .stops;
+    rel(bn.stops)["route"~"bus|train|tram|subway|light_rail"];
+    out;`;
```

Comment:
Can you please check why doesn't this work eg. [here](https://osmapp-git-fork-dlurak-publictransportlabels-zbycz.vercel.app/node/7040827488#14.99/50.0705/14.3826)?

btw, is this query used somewhere else? i wonder if it is proofed by time to be correct.

Reply:
Some trains don't hold at every train station, so I search actively for holding positions in 150 meters radius around the selected feature. The problem was that I only searched for `public_transport=stop_position` but this bus stop is tagged as `highway=bus_stop`.

The query is not proven by time yet, I wrote it by myself just yesterday.

Reply:
I see, thanks for the fix and explanation 🚀

Reply:
I see, thanks for the fix and explanation 🚀

Reply:
Fails when two different stops are closer than 150 meters:
https://osmapp.org/node/3377755042
https://osmapp.org/node/3377755037

Maybe distinguish the stops by name?

Reply:
Good idea! But I am not sure if all stop positions have a name to them.

Reply:
I've implemented and tested this idea on my local machine, and it works well for stations where everything is mapped correctly, such as *"Frankfurt am Main Hauptbahnhof."* However, I've encountered a problem with stations that have different names for the stop positions compared to the train station itself. Here are some examples:

| Train Station       | Stop Positions     |
| ------------------- | ------------------ |
| Friedberg (Hess)    | Friedberg (Hessen) |
| Brno hlavní nádraží | Brno hl.n.         |
| Wien Hauptbahnhof   | Wien Hbf           |

In Germany, central stations are generally referred to as *"Hauptbahnhof"* and often abbreviated as *"Hbf."* Some major train stations may have multiple levels with different names (e.g., *"Berlin Hauptbahnhof"* and *"Berlin Hauptbahnhof (tief)"*). However, I believe that separating their train lines might not be beneficial. There are also other cases where this approach isn't ideal.

Reply:
I like this conservative approach. If the _Brno hlavní nádraží_ (Public transport) and _Brno hl.n._ (Trains) will be separated it is more acceptable than having trains on the different stops (_Úzká_). Even the completely opposite approach - lines only from this stop position I'd rather see than the faulty departures. _But that's just my opinion!_

Theoretically, the stops should be in a relation [stop_area](https://wiki.openstreetmap.org/wiki/Tag:public_transport%3Dstop_area).

Reply:
I am not sure we are understanding each other correctly.
I created a new PR for the updates #180. What I mean is that when the node for the train station is called *Brno hlavní nádraží* but the name associated with the stop positions is *Brno hl.n.* the routes can't be clearly associated with the train station. So no routes will be shown, the vercel preview doesn't work but trust me it is the case. With the approach from #180 it will fail especially often for German central stations. Also it may become a problem with big bus stations.

The stop area relation seems interesting, do you know how well it is mapped in the real world

Reply:
I agree with @kudlav, that I would rather see different numbers on train_station and tram_station (if they have slightly different name). Then to be able to see wrong data.

The stop_area seems like a correct approach, it looks fairly used - [taginfo](https://taginfo.openstreetmap.org/tags/public_transport=stop_area).